providers/fips/self_test_kats.c

```diff
@@ -686,13 +686,21 @@ static int self_test_kas(OSSL_SELF_TEST *st, OSSL_LIB_CTX *libctx)
     return ret;
 }
 
+// 140-3: Need to do KAT, so need to control the DRBG
+static int set_kat_drbg(OSSL_LIB_CTX *ctx);
+static void reset_original_drbg(OSSL_LIB_CTX *ctx);
+
 static int self_test_signatures(OSSL_SELF_TEST *st, OSSL_LIB_CTX *libctx)
 {
     int i, ret = 1;
 
-    for (i = 0; i < (int)OSSL_NELEM(st_kat_sign_tests); ++i) {
-        if (!self_test_sign(&st_kat_sign_tests[i], st, libctx))
-            ret = 0;
+    // 140-3: Substitute and reset the private DRBG instance
+    if (set_kat_drbg(libctx)) {
+        for (i = 0; i < (int)OSSL_NELEM(st_kat_sign_tests); ++i) {
```

Comment:
This seems to bypass the test (returning 1) if `set_kat_drbg` fails?

Reply:
Yeah, it should fail...

Reply:
Added a pile of error handling and recovery.